src/garage/tf/algos/npo2.py

```diff
@@ -0,0 +1,562 @@
+
+        Args:
+            itr (int): Iteration number.
+            samples_data (dict): Processed sample data.
+                See process_samples() for details.
+
+        """
+        policy_opt_input_values = self._policy_opt_input_values(samples_data)
+        logger.log('Computing loss before')
+        loss_before = self._optimizer.loss(policy_opt_input_values)
+        logger.log('Computing KL before')
+        policy_kl_before = self._f_policy_kl(*policy_opt_input_values)
+        logger.log('Optimizing')
+        self._optimizer.optimize(policy_opt_input_values)
+        logger.log('Computing KL after')
+        policy_kl = self._f_policy_kl(*policy_opt_input_values)
+        logger.log('Computing loss after')
+        loss_after = self._optimizer.loss(policy_opt_input_values)
+        tabular.record('{}/LossBefore'.format(self.policy.name), loss_before)
+        tabular.record('{}/LossAfter'.format(self.policy.name), loss_after)
+        tabular.record('{}/dLoss'.format(self.policy.name),
+                       loss_before - loss_after)
+        tabular.record('{}/KLBefore'.format(self.policy.name),
+                       policy_kl_before)
+        tabular.record('{}/KL'.format(self.policy.name), policy_kl)
+        pol_ent = self._f_policy_entropy(*policy_opt_input_values)
+        ent = np.sum(pol_ent) / np.sum(samples_data['valids'])
+        tabular.record('{}/Entropy'.format(self.policy.name), ent)
+        tabular.record('{}/Perplexity'.format(self.policy.name), np.exp(ent))
+
+        self._fit_baseline_with_data(samples_data)
+
+        ev = np_tensor_utils.explained_variance_1d(samples_data['baselines'],
+                                                   samples_data['returns'],
+                                                   samples_data['valids'])
+
+        tabular.record('{}/ExplainedVariance'.format(self.baseline.name), ev)
+        self._old_policy.model.parameters = self.policy.model.parameters
+
+    def _build_inputs(self):
+        """Build input variables.
+
+        Returns:
+            namedtuple: Collection of variables to compute policy loss.
+            namedtuple: Collection of variables to do policy optimization.
+
+        """
+        observation_space = self.policy.observation_space
+        action_space = self.policy.action_space
+
+        with tf.name_scope('inputs'):
+            if self.flatten_input:
+                obs_var = tf.compat.v1.placeholder(
+                    tf.float32,
+                    shape=[None, None, observation_space.flat_dim],
+                    name='obs')
+            else:
+                obs_var = observation_space.to_tf_placeholder(name='obs',
+                                                              batch_dims=2)
+            action_var = action_space.to_tf_placeholder(name='action',
+                                                        batch_dims=2)
+            reward_var = new_tensor(name='reward', ndim=2, dtype=tf.float32)
```

Comment:
can you switch these out for tf.compat.v1.placeholder?

Reply:
Sure